module3-crud-and-data-models/r1-schema-and-data-models/README.md

```diff
@@ -39,123 +31,70 @@ CREATE TABLE customer (
+The database schema is one that contains list of attributes and instructions to tell the database engine how data is organised whereas data model is a collection of conceptional tools for describing data, data-relationship and consistency constraints.
 
 ## Introduction to CRUD operations
 
 When we are building APIs, we want our models to provide four basic types of functionality. The model must be able to Create, Read, Update, and Delete resources. Computer scientists often refer to these functions by the acronym `CRUD`. A model should have the ability to perform at most these four functions in order to be complete.
 
 ### CRUD and REST
 
+### What is REST?
+
+REST is a set of architectural constraints. API developers can implement REST in a variety of ways. When a client request is made via a RESTful API, it transfers a representation of the state of the resource to the requester or endpoint. This information, or representation, is delivered in one of several formats, JSON is the most generally popular file format to use.
+
+### CRUD (CREATE, READ, UPDATE, DELETE)
+
 In a REST environment, CRUD often corresponds to the HTTP methods `POST`, `GET`, `PUT`, and `DELETE`, respectively. These are the fundamental elements of a persistent storage system.
 
 For example, imagine we are working with a system that is keeping track of meals and their corresponding prices for a restaurant. Let’s look at how we would implement CRUD operations.
 
-#### Create
+- #### Create
 
 To create resources in a REST environment, we most commonly use the HTTP POST method. POST creates a new resource of the specified resource type.
 Imagine that we are adding a new food item to the stored list of dishes for this restaurant, and the dish objects are stored in a dishes resource.
 
+Note: Use `POST` when you want to add a child resource under resources collection.
+
 Request:
 `POST http://www.myrestaurant.com/api/dishes/`
 
 We will need to send the `dish` data too.
+`
 
-Body -
-
-```json
-{
-  "dish": {
-    "name": "Avocado Toast",
-    "price": 8
-  }
-}
-```
-
-Response: Status Code - 201 (Created)
+- #### Read
 
-This creates a new item with a `name` value of `“Avocado Toast”` and a price value of 8. Upon successful creation, the server should return a header with a link to the newly-created resource.
+To read resources in a REST environment, we use the GET method. in practice, reading a resource shouldn't change any information - it should only retrieve it.
 
-#### Read
-
-To read resources in a REST environment, we use the GET method. Reading a resource should never change any information - it should only retrieve it.
+Note: technically, you can change data in a `GET` request but since we are creating a RESTful API, you shouldn't do that, and, in-general, changing data in a `GET` request would be confusing to the users of your api.
```

Comment:
```suggestion
To read resources in a REST environment, we use the GET method. Reading a resource should never change any information - it should only retrieve it. REST itself is more like a set of guidelines. Technically, you can change data in a `GET` request, but since we are creating a RESTful API, you shouldn't do that. Having a GET request that updates data in your database be confusing to the users of your API and violate the expected behavior of REST.
```